### PR TITLE
after both, actionToFunc

### PR DESCRIPTION
**Pull request by haitch**
After both would connect 2 async task as input (from future), and start a new asyncTask.

the new task will be executed when both input task finished.

actionToFunc helps convert a function that doesn't emit result to satisfy asynctask interface.

### CODE REVIEW

File: after_both.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package asynctask
+
+import "context"
+
+// AfterBothFunc is a function that has 2 input.
+type AfterBothFunc[T, S, R any] func(context.Context, *T, *S) (*R, error)
+
+// AfterBoth runs the function after both 2 input task finished, and will be feed with result from 2 input task.
```
Comment (biefy) [inline after_both.go:8]: typo: `feed` -> `fed`

File: after_both.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	return Start(ctx, func(fCtx context.Context) (*R, error) {
+		t, err := tskT.Result(fCtx)
+		if err != nil {
+			return nil, err
+		}
+
+		s, err := tskS.Result(fCtx)
+		if err != nil {
+			return nil, err
+		}
```
Comment (biefy) [inline after_both.go:19]: This suggests error from `taskT` always take priority as error.  If this is the expected behavior, it should be clearly stated.  Also,  when `taskT` errors out, `taskS` could still be running.  This is not truly **After Both**.

Comment (haitch) [reply after_both.go:19]: 1. if one inputTask failed, this afterBoth task should fail anyway, right?
2. until later we introduce some option to allow ignore partial error, or we still fail, but wait for both task.

anyway, I will updade comments

Comment (haitch) [reply after_both.go:20]: we have WaitAllOptions in wait_all.go, we could reuse here in this afterBoth in future PRs, and then adjust the behavior according based on the option.

**Pull request merged**